src/Sentry/SentrySdk.cs

```diff
@@ -52,17 +48,40 @@ internal static IHub InitHub(SentryOptions options)
         }
 
         // Initialize native platform SDKs here
-#if __MOBILE__
         if (options.InitNativeSdks)
         {
 #if __IOS__
             InitSentryCocoaSdk(options);
 #elif ANDROID
             InitSentryAndroidSdk(options);
+#elif NET8_0_OR_GREATER
+            if (AotHelper.IsNativeAot)
+            {
+                InitNativeSdk(options);
+            }
 #endif
         }
-#endif
-        return new Hub(options);
+
+        // We init the hub after native SDK in case the native init needs to adapt some options.
+        var hub = new Hub(options);
+
+        // Do the initial scope sync, if configured by the native integration.
+        if (options.NativeContextWriter is { } contextWriter)
+        {
+            hub.ConfigureScope((scope) =>
+            {
+                Task.Run(() => contextWriter.Write(scope)).ContinueWith(t =>
+                {
+                    if (t.Exception is not null)
+                    {
+                        options.DiagnosticLogger?.LogWarning(
+                            "Failed to synchronize scope to the native SDK: {0}", t.Exception);
+                    }
+                });
```

Comment:
Also I believe there's a continuation that only happens if the task failed. 
```suggestion
                }, TaskContinuationOptions.OnlyOnFaulted);
```